pynest/nest/tests/test_parrot_neuron.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+#
+# test_parrot_neuron.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+# This script tests the parrot_neuron in NEST.
+
+import nest
+import unittest
+
+
+@nest.check_stack
+class ParrotNeuronTestCase(unittest.TestCase):
+    """Check parrot_neuron spike repetition properties"""
+
+    def setUp(self):
+        nest.set_verbosity(100)
+        nest.ResetKernel()
+
+        # set up source spike generator, as well as parrot neurons
+        self.spike_time = 1.
+        self.source = nest.Create("spike_generator", 1, {"spike_times": [self.spike_time]})
+        self.parrot = nest.Create('parrot_neuron')
+        self.spikes = nest.Create("spike_detector")
+
+        # record source and parrot spikes
+        nest.Connect(self.source, self.spikes)
+        nest.Connect(self.parrot, self.spikes)
+
+        self.gid_source = nest.GetStatus(self.source)[0]["global_id"]
+        self.gid_parrot = nest.GetStatus(self.parrot)[0]["global_id"]
```

Comment:
Why so complicated? `nest.Create()` returns a tuple of GIDs, so `self.gid_source = self.source[0]` would do, if you need it at all.